src/main/java/com/aws/greengrass/componentmanager/builtins/S3Downloader.java

```diff
@@ -86,21 +91,25 @@ protected String getArtifactFilename() {
 
     @Override
     public void cleanup() throws IOException {
-
     }
 
-    @SuppressWarnings(
-            {"PMD.CloseResource", "PMD.AvoidCatchingGenericException", "PMD.AvoidRethrowingException"})
+    @SuppressWarnings({"PMD.CloseResource", "PMD.AvoidCatchingGenericException", "PMD.AvoidRethrowingException"})
     @Override
     protected long download(long rangeStart, long rangeEnd, MessageDigest messageDigest)
             throws InterruptedException, PackageDownloadException {
         String bucket = s3ObjectPath.bucket;
         String key = s3ObjectPath.key;
 
-        GetObjectRequest getObjectRequest = GetObjectRequest.builder().bucket(bucket).key(key)
-                .range(String.format(HTTP_RANGE_HEADER_FORMAT, rangeStart, rangeEnd)).build();
-        logger.atDebug().kv("bucket", getObjectRequest.bucket()).kv("s3-key", getObjectRequest.key())
-                .kv("range", getObjectRequest.range()).log("Getting s3 object request");
+        GetObjectRequest getObjectRequest = GetObjectRequest.builder()
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

Potential log injection detected. Ensure all untrusted input is properly sanitized before logging. Use parameterized logging or validate input against a allow list to prevent log injection vulnerabilities. Consider using a dedicated logging library&#39;s built-in sanitization features when available. Learn more - https://cwe.mitre.org/data/definitions/117.html